lib/dispatcher/client-h2.js

```diff
@@ -151,6 +156,29 @@ function resumeH2 (client) {
   }
 }
 
+function onHttp2SendPing (session) {
+  const state = session[kHTTP2SessionState]
+  if ((session.closed || session.destroyed) && state.ping.interval != null) {
+    clearInterval(state.ping.interval)
+    state.ping.interval = null
+    return
+  }
+
+  // If no ping sent, do nothing
+  session.ping(onPing.bind(session))
+
+  function onPing (err, duration) {
+    const client = this[kClient]
+
+    if (err != null) {
+      // TODO: should we drop the session?
+      client[kOnError](err)
```

Comment:
Thinkin' twice, I'm over the fence of we should fail rather than actually just emit an event

Reply:
I think we should drop the sesson if this error.